docs/clients/gnmi-ctl.rst

```diff
@@ -2,129 +2,196 @@
-    * ``name``: Supported values are defined in chassis config file ``dpdk_port_config.pb.txt``.
-    * ``qemu-hotplug-mode``: Defines the hotplug mode. Supported values are (add/ADD, del/DEL).
-    * ``qemu-socket-ip``: Defines IP address for qemu monitor socket.
-    * ``qemu-socket-port``: Defines port number for qemu monitor socket.
-    * ``qemu-vm-mac-address``: Defines MAC address for port hotplugged to qemu VM.
-    * ``qemu-vm-netdev-id``: Defines netdev ID for port hotplugged to qemu VM.
-    * ``qemu-vm-chardev-id``: Defines chardev ID for port hotplugged to qemu VM.
-    * ``native-socket-path``: Defines the native path for vhost-user socket on host.
-    * ``qemu-vm-device-id``: Defines device ID for port hotplugged to qemu VM.
+   Use ``sudo`` when issuing gnmi-ctl command if you are a non-root user.
+
+gnmi client connects to gRPC port 9339 opened by the server and sends a
+protobuf based message.  Refer to gnmi.proto and gnmi_ext.proto for more
+details on type of messages and services that are available for a gNMI
+client.  These documents can be found in ``stratum/proto`` directory.
+
+Each CLI format should be in line with the YANG parser available in server
+which is a tree based hierarchy.  Tree has ``interfaces`` as root-node,
+followed by the device type.  Each device can have multiple ports, where
+port-name acts as a KEY.  SET command can be executed to configure the
+ports and GET command to retrieve the information about previously
+configured ports.
+
+Set attributes for a vhost port
+-------------------------------
+
+.. code-block:: text
+
+   gnmi-ctl set "device:virtual-device,name:net_vhost0,device-type:VIRTIO_NET"
+
+   gnmi-ctl set "device:virtual-device,name:net_vhost0,port-type:LINK"
+
+   gnmi-ctl set "device:virtual-device,name:net_vhost0,host-name:host1,
+              device-type:VIRTIO_NET,queues:1,
+              socket-path:/tmp/vhost-user-0,port-type:LINK"
+
+   gnmi-ctl set "device:virtual-device,name:net_vhost0,host-name:host1,
+              device-type:VIRTIO_NET,queues:1,
+              pipeline-name:pipe,mempool-name:MEMPOOL0,mtu:2000,
+              socket-path:/tmp/vhost-user-0,packet-dir:host,port-type:LINK"
+
+Parameters are key:value pairs. Here, virtual-device is a sub-node that can
+hold multiple ports like net_vhost0, net_vhost1, etc. Each port accepts
+multiple configuration parameters. These configuration parameters are again
+a key:value pair, which can be either passed in single CLI command or multiple
+CLI commands.
+
+Parameter descriptions:
+
+* ``name``: Supported values are defined in chassis config file
+  ``dpdk_port_config.pb.txt``.
+* ``host-name``: can be any string.
+* ``queues``: number of queues to be configured by target backend.
+* ``socket-path``: socket path for vhost port.
+* ``port-type``: Defines the port type and supported value for vhost port is
```

Comment:
Suggestion for future improvements. Maybe parameter descriptions can be a table with 3 columns:

Parameter name, description, default value if any

Reply:
It's worth considering.